### ISSUE TITLE
get_layer function cannot detect submodel's layers by name

### ISSUE THREAD

**Issue by nuxlear** [keras-team/keras#9590]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [ ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).
_( I use Tensorflow 1.5.0, but this issue may be no dependency from Tensorflow. )_

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

---

I use **keras.model.get_layer()** for my keras ML model, and I stuck by getting layer from submodel of model. 

My model have **input layer and two models** in its layer list, and both of two model have several conv and batchnorm layers. Anyway, I did try to get output layer of first submodel, but when I use get_layer() in my model, **it can access only 3 layers of input layer and two models, not submodel's.** So I have to call get_layer() twice, first get submodel from my model, and get wanted layer from this submodel. 

So I think that **when get_layer() function received name string, the function which can search layers recursively toward submodels to find matched layer by name, is more useful than now.** I have no idea of index problems but I think we can make get_layer() function like that when key is name. 

Let me know about other opinions or ideas of my suggestion. Thank you.

**Comment by tchaton** [comment]
You have to code a recursive function. But this think should be corrected in keras.
